tools/clone_flutter.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+set -e
+set -x
+
+if [ -z $ENGINE_PATH ]
+then
+  echo "This script is aimed to be run on CI environments. Do not run locally."
+  exit 1
+fi
+
+# Go to the engine git repo to get the date of the latest commit.
+cd $ENGINE_PATH/src/flutter
+
+# Special handling of release branches.
+ENGINE_BRANCH_NAME=`git branch | grep '*' | cut -d ' ' '-f2'`
+versionregex="v[[:digit:]]+\.{1}"
+ON_RELEASE_BRANCH=false
+echo "Engine on branch $ENGINE_BRANCH_NAME"
+if [[ $ENGINE_BRANCH_NAME =~ $versionregex ]]
+then
+  echo "release branch $ENGINE_BRANCH_NAME"
+  ON_RELEASE_BRANCH=true
+fi
+
+# Get latest commit's time for the engine repo.
+# Use date based on local time otherwise timezones might get mixed.
+LATEST_COMMIT_TIME_ENGINE=`git log -1 --date=local --format="%ad"`
+echo "Latest commit time on engine found as $LATEST_COMMIT_TIME_ENGINE"
+
+# Do rest of the task in the root directory
+cd ~
+mkdir -p $FRAMEWORK_PATH
+cd $FRAMEWORK_PATH
+# Clone the Flutter Framework.
+git clone https://github.com/flutter/flutter.git
+cd flutter
+
+FRAMEWORK_BRANCH_NAME=`git branch | grep '*' | cut -d ' ' '-f2'`
+if [[ "$ON_RELEASE_BRANCH" = true && ENGINE_BRANCH_NAME != FRAMEWORK_BRANCH_NAME ]]
+then
+  echo "For a release framework and engine should be on the same version."
+  echo "Switching branches on Framework to from $FRAMEWORK_BRANCH_NAME to $ENGINE_BRANCH_NAME"
+  # Switch to the same version branch with the engine.
+  # If same version branch does not exits, fail.
+  SWITCH_RESULT=`git checkout $ENGINE_BRANCH_NAME` || true
```

Comment:
Wont `|| true` make this always true?

Reply:
Yes. If we don't put it, when there is an error (such as branch not found) the script automatically ends without printing out the logs I added.

When we use this if the branch exists the SWITCH_RESULT is a non empty string if it fails it is an empty string